mobile/src/main/java/org/openhab/habdroid/background/ItemUpdateWorker.kt

```diff
@@ -91,34 +90,64 @@
                     )
                     return@runBlocking Result.failure(buildOutputData(true, 500))
                 }
-                if (value == "TOGGLE") {
-                    value = determineOppositeState(item)
-                    mappedValue = value
+
+                val value = inputData.getValueWithInfo(INPUT_DATA_VALUE)!!
+                val valueToBeSent = mapValueAccordingToItemTypeAndValue(value, item)
+                val actualMappedValue = if (value.value != valueToBeSent) {
+                    valueToBeSent
+                } else {
+                    value.mappedValue.orDefaultIfEmpty(value.value)
                 }
+
                 val result = if (inputData.getBoolean(INPUT_DATA_AS_COMMAND, false)) {
                     connection.httpClient
-                        .post("rest/items/$itemName", value, "text/plain;charset=UTF-8")
+                        .post("rest/items/$itemName", valueToBeSent, "text/plain;charset=UTF-8")
                         .asStatus()
                 } else {
                     connection.httpClient
-                        .put("rest/items/$itemName/state", value, "text/plain;charset=UTF-8")
+                        .put("rest/items/$itemName/state", valueToBeSent, "text/plain;charset=UTF-8")
                         .asStatus()
                 }
-                Log.d(TAG, "Item '$itemName' successfully updated to value $value")
+                Log.d(TAG, "Item '$itemName' successfully updated to value $valueToBeSent")
                 if (showToast) {
+                    val label = inputData.getString(INPUT_DATA_LABEL).orDefaultIfEmpty(itemName)
                     applicationContext.showToast(
-                        getItemUpdateSuccessMessage(applicationContext, label, value, mappedValue!!))
+                        getItemUpdateSuccessMessage(applicationContext, label, value.value, actualMappedValue))
                 }
                 sendTaskerSignalIfNeeded(taskerIntent, true, result.statusCode, null)
                 Result.success(buildOutputData(true, result.statusCode))
             } catch (e: HttpClient.HttpException) {
-                Log.e(TAG, "Error updating item '$itemName' to value $value. Got HTTP error ${e.statusCode}", e)
+                Log.e(TAG, "Error updating item '$itemName'. Got HTTP error ${e.statusCode}", e)
                 sendTaskerSignalIfNeeded(taskerIntent, true, e.statusCode, e.localizedMessage)
                 Result.failure(buildOutputData(true, e.statusCode))
             }
         }
     }
 
+    private fun mapValueAccordingToItemTypeAndValue(value: ValueWithInfo, item: Item) = when {
+        value.value == "TOGGLE" && item.isOfTypeOrGroupType(Item.Type.Switch) -> determineOppositeState(item)
```

Comment:
We need to extend this check to also cover Dimmer and Rollershutter (and maybe also Contact?)

Reply:
Why not:

```suggestion
        value.value == "TOGGLE" -> determineOppositeState(item)
```
See https://github.com/openhab/openhab-android/commit/b619fa9b45a7b63213fc8b5be83a417b465d629e#r35360730

Reply:
I don't want to fall back to `ON` for people who want to sent the string `TOGGLE` to e.g. a `StringItem`.

Reply:
I adjusted this check so all items are toggled where we suggest the `TOGGLE` value. Currently Contact items cannot be toggled, but this may be a good addition.